app/script/storage/StorageService.js

```diff
@@ -399,6 +399,10 @@ z.storage.StorageService = class StorageService {
    * @returns {Promise} Resolves with the primary key of the persisted object
    */
   save(store_name, primary_key, entity) {
+    if (!entity) {
+      throw new z.storage.StorageError('Value is undefined');
```

Comment:
Storage error expects a known type. This will be mapped to 'Unknown storage error' instead.